### PR TITLE
bpo-39648: Expand math.gcd() and math.lcm() to handle multiple arguments.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue39648

### CODE REVIEW

File: Doc/library/math.rst [hunk 1]
```diff
@@ -126,23 +126,19 @@ Number-theoretic and representation functions
...
 
-.. function:: gcd(a, b)
+.. function:: gcd(*integers)
 
-   Return the greatest common divisor of the integers *a* and *b*.  If either
-   *a* or *b* is nonzero, then the value of ``gcd(a, b)`` is the largest
-   positive integer that divides both *a* and *b*.  ``gcd(0, 0)`` returns
-   ``0``.
+   Return the greatest common divisor of the specified integer arguments.
+   If any of arguments is nonzero, then the returned value is the largest
```
Comment (mdickinson) [inline Doc/library/math.rst]:
```suggestion
   If any of the arguments is nonzero, then the returned value is the largest
```

File: Doc/library/math.rst [hunk 2]
```diff
@@ -126,23 +126,19 @@ Number-theoretic and representation functions
...
-
-.. function:: lcm(a, b)
-
-   Return the least common multiple of integers *a* and *b*.  The value of
-   ``lcm(a, b)`` is the smallest nonnegative integer that is a multiple of
-   both *a* and *b*.  If either *a* or *b* is zero then ``lcm(a, b)`` is zero.
-
-   .. versionadded:: 3.9
+   .. versionchanged:: 3.9
+      Added support of arbitrary number of arguments. Formerly, only two
```
Comment (mdickinson) [inline Doc/library/math.rst]:
```suggestion
      Added support for an arbitrary number of arguments. Formerly, only two
```

File: Doc/library/math.rst [hunk 3]
```diff
@@ -210,6 +206,17 @@ Number-theoretic and representation functions
    .. versionadded:: 3.8
 
 
+.. function:: lcm(*integers)
+
+   Return the least common multiple of the specified integer arguments.
+   If all arguments are nonzero, then the returned value is the smallest
+   positive integer that is a multiple of all arguments.  If any of arguments
```
Comment (mdickinson) [inline Doc/library/math.rst]: agreed with "positive" rather than "nonnegative"; I don't know what I was thinking before, but I was wrong.

Comment (mdickinson) [inline Doc/library/math.rst]:
```suggestion
   positive integer that is a multiple of all arguments.  If any of the arguments
```

File: Modules/mathmodule.c [hunk 4]
```diff
@@ -826,37 +826,125 @@ m_log10(double x)
...
+    }
+    for (i = 1; i < nargs; i++) {
+        x = PyNumber_Index(args[i]);
+        if (x == NULL) {
+            Py_DECREF(res);
+            return NULL;
+        }
+        if (res == _PyLong_One) {
+            /* Fast path: just check arguments.
+               It is okay to use identity comparison here. */
```
Comment (mdickinson) [inline Modules/mathmodule.c:854]: agreed!

**Review by mdickinson — APPROVED** [review]
LGTM; a couple of suggested wording tweaks in the documentation.

**Comment by mdickinson** [comment]
Thanks, Serhiy!

**Comment by serhiy-storchaka** [comment]
Thank you for your review Mark!

**Pull request merged**